src/bonsai/bonsai/bim/module/drawing/svgwriter.py

```diff
@@ -775,19 +776,87 @@ def get_box_alignment_parameters(box_alignment):
             "text-anchor": text_anchor,
         }
 
-    def add_fill_bg(self, element: svgwrite.text.Text, copy: bool = True) -> svgwrite.text.Text:
-        # Useful since tspans and texts do not support "background-color"
-        # so we just add a filter. Have to do it in a separate tag to avoid blurry image.
-        if copy:
-            element = element.copy()
-        if hasattr(element, "xml"):
-            attrib = element.xml.attrib
-        elif isinstance(element, ET.Element):
-            attrib = element.attrib
-        else:  # assuming it's svgwrite.base.BaseElement
-            attrib = element.attribs
-        attrib["filter"] = "url(#fill-background)"
-        return element
+    def add_fill_bg(self, element: svgwrite.text.Text, text, font_size, copy: bool = True) -> svgwrite.text.Text:
+        """Adds a background rectangle for each text line based on text dimensions."""
+        # font_path = "c:/users/ryan schultz/appdata/local/microsoft/windows/fonts/century_gothic.ttf"
+
+
+        attribs = getattr(element, 'attribs', None) or getattr(element, 'attrib', {})
+        #TO DO: need to find a way to use the font designated in the preferences, and grab it from the OS
+        # from matplotlib import font_manager
+        # if font_name := bpy.context.scene.DocProperties.drawing_font:
+        #     font_files = font_manager.findSystemFonts(fontpaths=None, fontext='ttf')
+        #     for font_path in font_files:
+        #         font = font_manager.FontProperties(fname=font_path)
+        #         if font.get_name() == font_name:
+        #             font = ImageFont.truetype(font_path, int(font_size))
+            
+
+        font_path = os.path.join(bpy.context.scene.BIMProperties.data_dir, "fonts", "OpenGost Type B TT.ttf")
+        font = ImageFont.truetype(font_path, int(font_size))
+
+        # Create a temporary image to draw text and measure its size
```

Comment:
This is may be why we avoided the rectangle approach - since we couldn't find a reliable way to connect rectangle to the text size using svg. The idea is to delegate everything to the svg renderer - because reimplementing the renderer is hard and calculating text box sizes manually, e.g. using some different rendered like `PIL`, may always have artifacts on some fonts/font sizes and will require magic numbers to adjust them.